dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -256,6 +258,30 @@ object Datadog {
         }
     }
 
+    // Executes all the pending queues in the upload/persistence executors.
+    // Tries to send all the granted data for each feature and then clears the folders.
+    // You should not use this method in production code. This is mainly for test purposes.
+    @Suppress("unused")
+    private fun flushStoredData() {
+        if (initialized.get()) {
+            val tasks = ArrayList<Runnable>()
+            (CoreFeature.persistenceExecutorService as? ThreadPoolExecutor)
+                ?.queue
+                ?.drainTo(tasks)
+            // we make sure we upload the currently locked files
+            CoreFeature.uploadExecutorService
+                .queue
+                .drainTo(tasks)
+            tasks.forEach {
+                it.run()
+            }
```

Comment:
This whole process could actually be a method of CoreFeature, e.g. `CoreFeature.drainAndStopExecutors`

Reply:
Agree, but maybe for now I will just say : `drainExecutors`